### ISSUE TITLE
SEC-47: Support PKCE in liberty oidc server

### ISSUE THREAD

**Issue by arunavemulapalli** [OpenLiberty/open-liberty#8480]
Add support to handle code_verifier and code_challenge in authorization and token requests. 
Corresponding AHA item is SEC-47.

**Comment by c00crane** [comment]
FAT:  https://github.ibm.com/was-liberty/ws-cd-open/issues/18009

**Comment by skasund** [comment]
STE approval complete.  Thanks for the slides @arunavemulapalli

**Comment by atosak** [comment]
@skaund, Is there a reason to have both deferred and approved tags for STE? It seems that the slides are uploaded.

**Comment by skasund** [comment]
@atosak  I must have accidentally clicked the deferred one when adding the labels.

**Comment by atosak** [comment]
Thanks for the confirmation.

**Comment by gscottj** [comment]
This feature does not have a UI or change any existing UIs. As such, there are no accessibility compliance requirements.  Accessibility approval granted.

**Comment by chirp1** [comment]
Hi! Chris Crane indicated that feature has new configuration attributes that will be self documenting through the metadata. So, there is nothing for ID to document. I'm approving the feature.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by dave-waddling** [comment]
**FAT Focal Approval comment:** Hi, I've finished reviewing this feature and all looks good. The testing passed cleanly in the latest SOE apart from hitting a known, infrequent, intermittent test bug. I'm therefore adding the FAT Focal Approval at this time.

**Comment by c00crane** [comment]
@marikaj123 The feature added 2 or 3 messages and new config options.  All of the messages and text for the options were merged before the Aug 16 deadline.

**Comment by marikaj123** [comment]
Please check if the returned translated messages made it into the release. Thank you.

**Comment by c00crane** [comment]
@marikaj123 - I just found translated messages and descriptions for all languages.

**Comment by c00crane** [comment]
Everything has been delivered, and approved - closing!
